cdc/sink/codec/canal_flat.go

```diff
@@ -52,21 +51,19 @@ func NewCanalFlatEventBatchEncoder() EventBatchEncoder {
 }
 
 type canalFlatEventBatchEncoderBuilder struct {
-	opts map[string]string
+	config *Config
 }
 
 // Build a `CanalFlatEventBatchEncoder`
-func (b *canalFlatEventBatchEncoderBuilder) Build(_ context.Context) (EventBatchEncoder, error) {
+func (b *canalFlatEventBatchEncoderBuilder) Build(_ context.Context) EventBatchEncoder {
```

Comment:
```suggestion
func (b *canalFlatEventBatchEncoderBuilder) Build() EventBatchEncoder {
```

Could you remove `Context`? It looks unnecessary.